File: packages/shared-ts/src/content-type.ts
export function isTextContentType(contentType: string): boolean {
  if (!contentType) return false;
  const normalized = contentType.toLowerCase();
  return (
    normalized.startsWith('text/') ||
    normalized.includes('json') ||
    normalized.includes('xml') ||
    normalized.includes('javascript') ||
    normalized.includes('ecmascript') ||
    normalized.includes('html') ||
    normalized.includes('css') ||
    normalized.includes('svg')
  );
}

export function isFormContentType(contentType: string): boolean {
  if (!contentType) return false;
  return contentType.toLowerCase().includes('urlencoded');
}

export function isTextRequestContentType(contentType: string): boolean {
  if (!contentType) return false;
  return isTextContentType(contentType) || isFormContentType(contentType);
}

File: packages/swift-server/Sources/Server/ContentType.swift
import Foundation






















func isTextContentType(_ contentType: String) -> Bool {
    if contentType.isEmpty { return false }
    let normalized = contentType.lowercased()
    return normalized.hasPrefix("text/")
        || normalized.contains("json")
        || normalized.contains("xml")
        || normalized.contains("javascript")
        || normalized.contains("ecmascript")
        || normalized.contains("html")
        || normalized.contains("css")
        || normalized.contains("svg")
}






func isFormContentType(_ contentType: String) -> Bool {
    if contentType.isEmpty { return false }
    return contentType.lowercased().contains("urlencoded")
}




















func isTextRequestContentType(_ contentType: String) -> Bool {
    if contentType.isEmpty { return false }
    return isTextContentType(contentType) || isFormContentType(contentType)
}
